google/cloud/bigquery/v2/minimal/internal/dataset.cc

```diff
@@ -174,6 +175,172 @@ void from_json(nlohmann::json const& j, Dataset& d) {
+      .Field("target_types", target_types)
+      .Build();
+}
+
+std::string Access::DebugString(absl::string_view name,
+                                TracingOptions const& options,
+                                int indent) const {
+  return internal::DebugFormatter(name, options, indent)
+      .StringField("role", role)
+      .StringField("domain", domain)
+      .StringField("special_group", special_group)
+      .StringField("iam_member", iam_member)
+      .SubMessage("view", view)
+      .SubMessage("routine", routine)
+      .SubMessage("dataset", dataset)
+      .Build();
+}
+
+std::string HiveMetastoreConnectivity::DebugString(
+    absl::string_view name, TracingOptions const& options, int indent) const {
+  return internal::DebugFormatter(name, options, indent)
+      .StringField("access_uri_type", access_uri_type)
+      .StringField("access_uri", access_uri)
+      .StringField("metadata_connection", metadata_connection)
+      .StringField("storage_connection", storage_connection)
+      .Build();
+}
+
+std::string HiveDatabaseReference::DebugString(absl::string_view name,
+                                               TracingOptions const& options,
+                                               int indent) const {
+  return internal::DebugFormatter(name, options, indent)
+      .StringField("catalog_id", catalog_id)
+      .StringField("database", database)
+      .SubMessage("metadata_connectivity", metadata_connectivity)
+      .Build();
+}
+
+std::string ExternalDatasetReference::DebugString(absl::string_view name,
+                                                  TracingOptions const& options,
+                                                  int indent) const {
+  return internal::DebugFormatter(name, options, indent)
+      .SubMessage("hive_database", hive_database)
+      .Build();
+}
+
+std::string GcpTag::DebugString(absl::string_view name,
+                                TracingOptions const& options,
+                                int indent) const {
+  return internal::DebugFormatter(name, options, indent)
+      .StringField("tag_key", tag_key)
+      .StringField("tag_value", tag_value)
+      .Build();
+}
+
+std::string Dataset::DebugString(absl::string_view name,
+                                 TracingOptions const& options,
+                                 int indent) const {
+  return internal::DebugFormatter(name, options, indent)
+      .StringField("kind", kind)
+      .StringField("etag", etag)
+      .StringField("id", id)
+      .StringField("self_link", self_link)
+      .StringField("friendly_name", friendly_name)
+      .StringField("description", description)
+      .StringField("type", type)
+      .StringField("location", location)
+      .StringField("default_collation", default_collation)
+      .Field("published", published)
+      .Field("is_case_insensitive", is_case_insensitive)
+      .Field("default_table_expiration", default_table_expiration.count())
```

Comment:
It looks like we need some formatter support for `std::chrono::duration`.  I'll do something for that, and then loop back here.